jcl/src/java.base/share/classes/java/lang/invoke/CollectHandle.java

```diff
@@ -99,10 +140,16 @@ protected final ThunkTuple computeThunks(Object arg) {
 		return thunkTable().get(new ThunkKeyWithIntArray(ThunkKey.computeThunkableType(type()), collectArguments));
 	}
 
-	private final Object allocateArray() {
+	private static final Object allocateArray(int size, Class<?> componentType) {
```

Comment:
This is only called from one place - `createEmptyArray()` - can you inline the implementation there?  It would also be cleaner to add an `import java.lang.reflect.Array;` line and call it as `Array.newInstance(...)`.

Reply:
This separate to facilitate special casing by the JIT. Some kind of comment to that affect is probably warranted but isolating this use case into a function we can recognize seems a reasonable design choice to me.

Reply:
@andrewcraik that makes sense for `allocateArray(MethodHandle)`.  This copy is only called in the ctor for CollectHandle and doesn't need jit special casing.

Reply:
@DanHeidinga Thanks for pointing this out. I fixed it.